<launch>

    <include file="$(find voxel_trajectory)/launch/outer_simulation.launch"/>


    <node 
 	    pkg 	= "voxel_trajectory"
	    type 	= "test_limited_sensing_node"
 	    name 	= "limited_sensor"
	    output  = "screen">
        <param  name = "limited_sensing/sensing_raduis"     value = "3.0"/>
        <param  name = "limited_sensing/sensing_resolution" value = "0.1"/>
        <param  name = "limited_sensing/sensing_rate"       value = "5"/>
        <param  name = "limited_sensing/obstacle_file"
                    value = "$(find voxel_trajectory)/data/test_case_2/obstacle.txt" />

        <remap  from = "~odometry"              to = "/state_ukf/odom" />
        <remap  from = "~sensed_points"         to = "/trajectory_generator/obstacle_points"/>
	<remap  from = "~obstacle_points"       to = "~obstacle_points"/>
	<remap  from = "~obstacle_blocks"       to = "~obstacle_blocks"/>
    </node>

    <node
        pkg     = "voxel_trajectory"
        type    = "trajectory_server_node"
        output  = "screen"
        name    = "trajectory_generator">

        <param  name = "init/visualization" value = "true"/>

        <param  name = "map/boundary/lower_x"  value = "0.0"/>
        <param  name = "map/boundary/upper_x"  value = "10.24"/>
        <param  name = "map/boundary/lower_y"  value = "0.0"/>
        <param  name = "map/boundary/upper_y"  value = "10.24"/>
        <param  name = "map/boundary/lower_z"  value = "0.0"/>
        <param  name = "map/boundary/upper_z"  value = "10.24"/>
        <param  name = "map/resolution"  value = "0.05"/>

        <param  name = "max_velocity"  value = "3.0"/>
        <param  name = "max_acceleration"  value = "3.0"/>

        <!-- subscribed topics -->
        <remap  from = "~obstacle_points"   to = "~obstacle_points"/>
        <remap  from = "~obstacle_blocks"   to = "~obstacle_blocks"/>
        <remap  from = "~odometry"          to = "/state_ukf/odom"/>
        <remap  from = "~goal_point"        to = "~goal_point"/>
        <remap  from = "~waypoints"         to = "/waypoint_generator/waypoints"/>

        <!-- advertised topics -->
        <remap  from = "~desired_state"     to = "/position_cmd"/>

        <!-- advertised visualization topics -->
        <remap  from = "~map_vis"           to = "/trajectory_generator/map_points"/>
        <remap  from = "~trajectory_vis"    to = "/trajectory_generator/line_strip"/>
        <remap  from = "~path_vis"          to = "/trajectory_generator/voxel_path"/>
        <remap  from = "~inflated_path_vis" to = "/trajectory_generator/inflated_path"/>
        <remap  from = "~checkpoints_vis"   to = "/trajectory_generator/check_point"/>
    </node>

</launch>
